src/main.rs

```diff
@@ -250,10 +287,19 @@ impl SendState {
             borsh::to_vec(&header)?.len()
         );
 
+        let in_fd = if let SendStateInner::InProgress {
```

Comment:
ugh, I don't like this. We know that state is guaranteed to be `InProgress` at this point. Is there a nicer way to do this?

Reply:
Since you already do the match above, in addition to returning `offset` from it, we could return `file.as_raw_fd()` from it.